src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectReferenceFileService.java

```diff
@@ -0,0 +1,126 @@
+package ca.corefacility.bioinformatics.irida.ria.web.services;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Locale;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+import ca.corefacility.bioinformatics.irida.exceptions.EntityNotFoundException;
+import ca.corefacility.bioinformatics.irida.exceptions.UnsupportedReferenceFileContentError;
+import ca.corefacility.bioinformatics.irida.model.project.Project;
+import ca.corefacility.bioinformatics.irida.model.project.ReferenceFile;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.dto.ajax.AjaxResponse;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.dto.ajax.AjaxUpdateItemSuccessResponse;
+import ca.corefacility.bioinformatics.irida.service.ProjectService;
+import ca.corefacility.bioinformatics.irida.service.ReferenceFileService;
+
+/**
+ * A utility class for formatting responses for the project reference files page UI.
+ */
+
+@Component
+public class UIProjectReferenceFileService {
+	private static final Logger logger = LoggerFactory.getLogger(UIProjectReferenceFileService.class);
+
+	private final ProjectService projectService;
+	private final ReferenceFileService referenceFileService;
+	private final MessageSource messageSource;
+
+	@Autowired
+	public UIProjectReferenceFileService(ProjectService projectService, ReferenceFileService referenceFileService, MessageSource messageSource) {
+		this.projectService = projectService;
+		this.referenceFileService = referenceFileService;
+		this.messageSource = messageSource;
+	}
+
+	/**
+	 * Add a new reference file to a project.
+	 *
+	 * @param projectId The id of the project to add the file to.
+	 * @param files     {@link List} of {@link MultipartFile} file being uploaded.
+	 * @param locale    locale of the logged in user
+	 * @return Return success or error message if file was successfully uploaded or not
+	 * @throws UnsupportedReferenceFileContentError if content is invalid
+	 * @throws IOException if there is an I/O error
+	 */
+	public AjaxResponse addReferenceFileToProject(Long projectId, List<MultipartFile> files, final Locale locale) throws UnsupportedReferenceFileContentError, IOException {
```

Comment:
Your controllers should return the ajax response not your services.

Reply:
Updated in 219a5fbdd6a0ee0e31fd61c6c443e0020191c82d